Создайте новый файл или откройте active в системном пути нашего приложения

opjs105rogue opened the pull request:
Hello, @agoalofalife! Add IIFE and now, when the user opens the application, a folder is created where the user's text document will be stored.

This is only the first step, interaction with the txt file through the application will be a little later

opjs105rogue committed:
Add IIFE for create new path and file if are not

opjs105rogue committed:
Add message with a greeting if the user has opened the application for the first time

opjs105rogue commented:
add a welcome message if the user opened the application for the first time (that is, he does not yet have a folder where the file will be stored) I connected it with a common logic

opjs105rogue committed:
Merge branch 'main' into PathSystemFile

opjs105rogue commented:
I merge my file with main rep.

opjs105rogue committed:
Fix tray icon with new main file's code

opjs105rogue commented:
@agoalofalife привет! В общем, я скопировал еще не замерженный код Гамзата, чтобы не терять время, все равно пришлось бы мержить, в нем Гамзат установил логики взаимодействия с трей иконкой для обоих окон = окна с паролем и основного, где вводится текст, но у меня оно работало некорректно, я добавил/подорректировал логику закрытия приложения через конекстное меню и взаимодействие через клики по иконке в трее, так как у меня вылетала ошибка, что переменная не определена (не могу вспомнить какая)
Посмотри пожалуйста, работает ли все корректно, если все ок, нужно замержить этот файл, так как у всех он должен работать корректно в плане взаимодейтсвия с трей иконкой и он основан на последнем файле Гамзата. И уже с него снова стартану выполнять задачу

opjs105rogue committed:
Add IIFE for create new path and file if are not and Add message with a greeting if the user has opened the application for the first time

opjs105rogue commented:
Hi, @agoalofalife ! Added the creation of a butt and a file to the new code. Now, if our application is delivered without a file, it will be created. And if it is already with the file, nothing will happen, but as soon as the user deletes the folder and launches the application, it will create a folder with the file for him and inform him about it before entering the password. You can test it, it should work

agoalofalife commented on main.js:
```diff
@@ -1,158 +1,175 @@
 
-  // Only open DevTools in local environment!
-  if (process.env.APP_ENV === 'local') {
+  if (process.env.APP_ENV === "local") {
```
Why did you decide use Double quotes?

opjs105rogue replied:
When I incorrectly returned the function when creating the tray icon, you fixed my mistakes, including single commas in some places, so I thought "yes, indeed, this will look more consistent in the code." Therefore, I started using them whenever possible

agoalofalife commented on main.js:
```diff
@@ -1,158 +1,175 @@
+  });
+
+  // Path for dir with txt file
+  const userFilesDir = path.join(__dirname, "User files");
```
First of all, i think we have to out this path in env variable (i.e. file environment)
`USER_FILE_PATH=/files/text.txt`
Obviously that path is relative

agoalofalife replied:
and also, I think this is a strange name for directory, never use with space

RGamz replied:
agree with Ilya, please change dir name to "user_files" for example, no spaces.

agoalofalife commented on main.js:
```diff
@@ -1,158 +1,175 @@
+      fs.writeFileSync(filePath, "", "utf8");
+
+    // Check console
+      console.log(`File created at: ${filePath}`);
```
ok, I think it has to be info level

agoalofalife replied:
And I don't know why, but I haven't seen these logs anywhere

opjs105rogue replied:
the information should be displayed when creating a file in the editor console, and not in the electron application, if it is not displayed, please let me know
![Снимок экрана от 2024-10-21 15-58-40](https://github.com/user-attachments/assets/4b8c76b8-bd76-48f2-8724-d9c9bb06fed7)

agoalofalife commented on main.js:
```diff
@@ -1,158 +1,175 @@
+    if (!fs.existsSync(filePath)) {
+      fs.writeFileSync(filePath, "", "utf8");
+
+    // Check console
```
comment is redundant

agoalofalife commented on main.js:
```diff
@@ -1,158 +1,175 @@
+      fs.mkdirSync(userFilesDir); 
+    }
+
+    //Abd add file 'user_file', if its not
```
comment is redundant

agoalofalife commented on main.js:
```diff
@@ -1,158 +1,175 @@
+  // Path for file in dir and add const
+  const filePath = path.join(userFilesDir, "user_file.txt");
+
+  // add dir 'User files', if its not
```
comment is redundant

agoalofalife commented on main.js:
```diff
@@ -1,158 +1,175 @@
+  // Path for dir with txt file
+  const userFilesDir = path.join(__dirname, "User files");
+
+  // Path for file in dir and add const
```
comment is redundant

agoalofalife commented on main.js:
```diff
@@ -1,158 +1,175 @@
+        type: "info"
+      });
+    } else {
+    // Check console
```
comment is redundant

RGamz replied:
did not get this message one the first run of the app...

agoalofalife commented on main.js:
```diff
@@ -1,158 +1,175 @@
+      });
+    } else {
+    // Check console
+      console.log(`File already exists at: ${filePath}`);
```
I'm not sure that is a useful info, we will see this log every time after the file will be created

agoalofalife commented on main.js:
```diff
@@ -1,158 +1,175 @@
+      console.log(`File created at: ${filePath}`);
+
+      //Add message with a greeting if the user has opened the application for the first time
+      dialog.showMessageBox(welcomeView, {
```
might be - but first user has to enter password

RGamz commented on main.js:
```diff
@@ -1,158 +1,175 @@
-  if (BrowserWindow.getAllWindows().length === 0) createWindow();
+  app.on('activate', () => {
+    if (BrowserWindow.getAllWindows().length === 0) renderPasswordWindow();
+  });
```
i am not sure why but for me on mac the app immediately renders the main window, not password window. Normally the file creation should happen after creating and entering the password. please check if it works for you. checking on mac.

opjs105rogue committed:
add env variable, changed place fir function with create file and clear cpmments

opjs105rogue commented:
@agoalofalife  @RGamz  Hi! 
Renamed the folder without a space.
Created an env variable and use it with function to create folder.
Deleted unnecessary comments.
Changed the location of the function with the creation of the file, now a folder with the file is created when the main window opens

agoalofalife commented on main.js:
```diff
@@ -1,158 +1,167 @@
+  });
+  
+  // Path for dir with txt file
+  const userFilesDir = path.dirname(process.env.USER_FILE_PATH);
```
I think we have add error log if env variable has not been pointed in env file

agoalofalife commented on main.js:
```diff
@@ -1,158 +1,167 @@
 import dotenv from "dotenv";
+import fs from "fs";
 
 const __filename = fileURLToPath(import.meta.url);
```
It uses anywhere?

opjs105rogue committed:
fix click\double-click\right-click function for OC

opjs105rogue committed:
add error log and emergency exit from the application in case of an error

opjs105rogue committed:
delete variable __filename and fix conflicts with main branch

opjs105rogue commented:
Hi! @agoalofalife!

Fixed the mouse click issue on macOS (at least I think so). I also corrected the opening of the application through the tray icon (now it opens with a double click, and on Linux with one click due to the operation of the Linux system).

Added error output to the text editor console if the environment variable in .env is not set. The application will not start if it is not specified, and it is visible in the console.

Deleted an unnecessary unused variable and fixed conflicts with the main branch after a recent merge.

Take a look at this and if any errors occur, please let me know

!!!! I forgot to delete the folder that is created when opening the application, now you will not see a message about creating a file, but if you delete the folder in your project, the program will create a directory with the file itself and write about it in the notification!!!!!

agoalofalife commented on main.js:
```diff
@@ -47,6 +52,30 @@ const renderMainWindow = () => {
+  })();
 };
 
 const renderPasswordWindow = () => {
```
where we are using these `const { width, height }`  variables?

agoalofalife commented on user_files/user_file.txt:
we have to add this folder in .gitignore - because this file has sensitive data and can't be added in git

agoalofalife commented on main.js:
```diff
@@ -90,15 +119,17 @@ app.whenReady().then(() => {
-    if (mainView) { // Check if mainView exists
-      if (mainView.isVisible()) {
-        mainView.hide();
+  tray.on(process.platform === "lunux" ? "click" : "double-click", () => {
```
lunux - maybe linux? i've already tried but it didn't help with tray

opjs105rogue committed:
delete another unnecessary unused variables and add user's path in gitignore

opjs105rogue committed:
Chage function for click on system

opjs105rogue commented:
@agoalofalife ,hi!
I have made changes according to your comments.

Changed the function with opening/hiding the app and opening the tray. Now the function looks more simple. I also separated the left click separately (to open/hide the application) and the right click to open the context menu as it was before (I read that it would be more reliable this way). 
+ tweaked the popUpContextMenu method
Lets try this, if it wont work,  please let me know

agoalofalife committed:
delete excesive log
